x-pack/plugins/apm/server/routes/__test__/routeFailures.test.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Server } from 'hapi';
+// @ts-ignore
+import { initErrorsApi } from '../errors';
+import { initServicesApi } from '../services';
+// @ts-ignore
+import { initStatusApi } from '../status_check';
+import { initTracesApi } from '../traces';
+import { initTransactionsApi } from '../transactions';
+
+async function testRouteFailures(init: (server: Server) => void) {
+  const mockServer = { route: jest.fn() };
+  init((mockServer as unknown) as Server);
+  expect(mockServer.route).toHaveBeenCalled();
+
+  const routes = mockServer.route.mock.calls;
+  const mockReq = {
+    params: {},
+    query: {},
+    pre: {
+      setup: {
+        config: { get: jest.fn() },
+        client: jest.fn(() => Promise.reject(new Error('request failed')))
+      }
+    }
+  };
+
+  routes.forEach(async (route, i) => {
+    test(`route ${i + 1} of ${
+      routes.length
+    } should fail with a Boom error`, async () => {
+      try {
+        await route[0].handler(mockReq);
+      } catch (err) {
+        expect(err.isBoom).toEqual(true);
+        return;
+      }
+
+      throw new Error('Route succeeded when it should have failed');
```

Comment:
There is a built-in for this:

```
  expect(() => route[0].handler(mockReq)).toThrowError(BoomError);
```

There are different matchers. You might have to modify it a bit to work.

Reply:
Yeah, unfortunately the ones I tried didn't work here because I want to test things about the rejected error. Where does `BoomError` come from in your example?